src/modules/hip/kernel/gamma_correction.hpp

```diff
@@ -0,0 +1,274 @@
+    uint dstIdx = (id_z * nStrideDst) + (id_y * hStrideDst) + id_x;
+
+    d_float8 src_f8, dst_f8;
+
+    rpp_hip_load8_and_unpack_to_float8(srcPtr, srcIdx, &src_f8);
+    gamma_correction_hip_compute(srcPtr, &src_f8, &dst_f8, gamma[id_z]);
+    rpp_hip_pack_float8_and_store8(dstPtr, dstIdx, &dst_f8);
+
+    if (channelsDst == 3)
+    {
+        srcIdx += cStrideSrc;
+        dstIdx += cStrideDst;
+
+        rpp_hip_load8_and_unpack_to_float8(srcPtr, srcIdx, &src_f8);
+        gamma_correction_hip_compute(srcPtr, &src_f8, &dst_f8, gamma[id_z]);
+        rpp_hip_pack_float8_and_store8(dstPtr, dstIdx, &dst_f8);
+
+        srcIdx += cStrideSrc;
+        dstIdx += cStrideDst;
+
+        rpp_hip_load8_and_unpack_to_float8(srcPtr, srcIdx, &src_f8);
+        gamma_correction_hip_compute(srcPtr, &src_f8, &dst_f8, gamma[id_z]);
+        rpp_hip_pack_float8_and_store8(dstPtr, dstIdx, &dst_f8);
+    }
+}
+
+template <typename T>
+__global__ void gamma_correction_pkd3_pln3_tensor(T *srcPtr,
+                                                  int nStrideSrc,
+                                                  int hStrideSrc,
+                                                  T *dstPtr,
+                                                  int nStrideDst,
+                                                  int cStrideDst,
+                                                  int hStrideDst,
+                                                  float *gamma,
+                                                  RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * nStrideSrc) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * hStrideSrc) + ((id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3);
+    uint dstIdx = (id_z * nStrideDst) + (id_y * hStrideDst) + id_x;
+
+    d_float24 src_f24, dst_f24;
+
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr, srcIdx, &src_f24);
+    gamma_correction_hip_compute(srcPtr, &src_f24.x, &dst_f24.x, gamma[id_z]);
+    rpp_hip_pack_float8_and_store8(dstPtr, dstIdx, &dst_f24.x);
+
+    dstIdx += cStrideDst;
+
+    gamma_correction_hip_compute(srcPtr, &src_f24.y, &dst_f24.y, gamma[id_z]);
+    rpp_hip_pack_float8_and_store8(dstPtr, dstIdx, &dst_f24.y);
+
+    dstIdx += cStrideDst;
+
+    gamma_correction_hip_compute(srcPtr, &src_f24.z, &dst_f24.z, gamma[id_z]);
+    rpp_hip_pack_float8_and_store8(dstPtr, dstIdx, &dst_f24.z);
+}
+
+template <typename T>
+__global__ void gamma_correction_pln3_pkd3_tensor(T *srcPtr,
+                                                  int nStrideSrc,
```

Comment:
can you pass all strides as one vector value instead? like int4 srcStride, dstStride

Reply:
I was thinking the index computation readability inside the kernel might be much better with independent variables. Since using an vector or array might need addressing with .x/.y/.z/.w or [0/1/2] indices. 

Like uint dstIdx = (id_z * nStrideDst) + (id_y * hStrideDst) + id_x;

In the above line, its easy to understand that we are jumping:
- to the next image with nStrideDst
- to the next row with hStrideDst

Also, in certain cases we only need n/c/h strides, and in others we need h/n strides - that varies too.